[Term Entry] Python built-in functions slice() #5117 (#5278)
Codecademy/docs: content/python/concepts/built-in-functions/terms/slice/slice.md

---
Title: '.slice()'
Description: 'Returns a slice object from a tuple.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags: 
  - 'Data Structures'
CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
  - 'learn-python-3'
  - 'paths/computer-science'
---

In Python, the .slice() function is used to create a slice object, used to slice any sequence( e.g. - string, tuple, list, etc).

## Syntax

```pseudo
slice(start, stop, step)
```

> where it takes three parameters:
> - `start:` This integer indicates where the slicing of the object begins and is optional. Default value is None if not provided. 
> - `stop:` This integer indicates up to where the slicing takes place. It stops at index stop-1 (the last element of the sequence). 
> - `step:` This integer value is used to fix the increment between each index for slicing and is optional. Its default value is null if not provided. 
> 

Reviewer: Modifications:

- Let's remove the introductory line as it's not needed
- Let's do some minor changes in the bulleted list

```suggestion
- `start`: This optional integer indicates where the slicing of the object begins. The default value is `None` if not provided.
- `stop`: This integer indicates up to where the slicing takes place. It stops at index `stop-1` (the last element of the sequence).
- `step`: This optional integer fixes the increment between each index for slicing. The default value is `None` if not provided.
```

## Example

> The following example shows the use of the `.slice()' function:

Reviewer: Let's do some minor changes:

```suggestion
The following example shows the use of the `slice()` function:
```

```py

> #create a tuple with some values in it
> fruits = ("Apple", "Banana", "Orange", "Kiwi", "Strawberry", "Plum", "Watermelon")
> 
> #create a slice object
> a = slice(3)
> 
> #return the slice result
> print(fruits[a])
> 

Reviewer: Let's start the comments using capital letters and do some minor changes:

```suggestion
# Create a tuple with some values in it
fruits = ("Apple", "Banana", "Orange", "Kiwi", "Strawberry", "Plum", "Watermelon")

# Create a slice object
a = slice(3)

# Return the resultant slice object
print(fruits[a])
```

> ```
> The output of the code is as follows:
> 
> ```shell
> ('Apple', 'Banana', 'Orange')
> 
> ```

Reviewer: Let's do some minor changes:

````suggestion
```

The output of the code is as follows:

```shell
('Apple', 'Banana', 'Orange')
```
````

> ## Codebyte Example (if applicable)
> 
> Run the following codebyte to understand how the `.slice()` is used to create a slice object from a given sequence(here, a string) between a defined range:
> 
> ```codebyte/python
> 
> # Example runnable code block.
> #string slicing
> my_string = 'CompetetiveProgramming'
> slice_obj1 = slice(5)
> slice_obj2 = slice(1, 8, 2)
> 
> print("Resulting slices:")
> print(my_string[slice_obj1])
> print(my_string[slice_obj2])
> 
> ```

Reviewer: Modifications:

- Let's remove 'if applicable' and the comments as they're not needed
- Let's do some minor changes

````suggestion
## Codebyte Example

The following codebyte example demonstrates how the `slice()` function is used to create a slice object from a given sequence (here, a string) between a defined range:

```codebyte/python
my_string = 'CompetitiveProgramming'
slice_obj1 = slice(5)
slice_obj2 = slice(1, 8, 2)

print("Resulting slices:")
print(my_string[slice_obj1])
print(my_string[slice_obj2])
```
````
